include/solvers/flow_control.h

```diff
@@ -34,21 +55,96 @@
 
 using namespace dealii;
 
+
+/**
+ * @brief FlowControl. The FlowControl class calculates the volumetric flow
+ * rate at the inlet or at a selected boundary. It also allows to dynamically
+ * control the flow with a beta coefficient calculated at each step time.
+ */
+
 template <int dim, typename VectorType>
 class FlowControl
 {
 public:
-  Tensor<1, dim>
-  calculate_beta(const DoFHandler<dim> &               dof_handler,
-                 const VectorType &                    present_solution,
-                 const Parameters::DynamicFlowControl &flow_control,
+  /**
+   * @brief calculate_flow_rate. This function calculates the volumetric flow
+   * rate at the selected boundary. It actually calculates the flow rate through
+   * the summation of the value at each cell surface with the normal vector,
+   * the velocity value and the area.
+   *
+   * @param dof_handler. The argument used to get finite elements.
+   *
+   * @param present_solution. The vector which contains all the values to
+   *                          calculate the flow rate.
+   *
+   * @param flow_control. The flow control parameters
+   *
+   * @param fem_parameters. The FEM parameters
+   *
+   * @param mpi_communicator. The mpi communicator information
+   */
+  void
+  calculate_flow_rate(const DoFHandler<dim> &               dof_handler,
+                      const VectorType &                    present_solution,
+                      const Parameters::DynamicFlowControl &flow_control,
+                      const Parameters::FEM &               fem_parameters,
+                      const MPI_Comm &                      mpi_communicator);
+
+  /**
+   * @brief calculate_beta. This function calculates a beta coefficient which
+   * applies a force to the flow in order to adjust the flow rate to a desired
+   * value through the previous flow rate. Once the flow rate is reached, the
+   * algorithm calculates a new beta to keep a constant flow rate.
+   *
+   * @param flow_control. The flow control parameters
+   *
+   * @param simulation_control. The simulation control parameters
+   *
+   * @param step_number. The current step
+   */
+  void
+  calculate_beta(const Parameters::DynamicFlowControl &flow_control,
                  const Parameters::SimulationControl & simulation_control,
```

Comment:
Why does calculate Beta require the parameter at every calculation?
You should pass the parameter in the constructor of the class and not at every call of calculate_beta. Otherwise it becomes superflous.

Reply:
Oh yeah, good idea!